server/src/main/java/org/opensearch/action/search/TransportSearchAction.java

```diff
@@ -290,6 +301,118 @@ long getAbsoluteStartMillis() {
+        }
+
+        public boolean isPhaseTookEnabled() {
+            return phaseTookEnabled;
+        }
+
+        SearchResponse.PhaseTook getPhaseTook() {
+            if (phaseTookEnabled) {
+                return new SearchResponse.PhaseTook(
+                    totalStats.dfsPreQueryTotal.count(),
+                    totalStats.canMatchTotal.count(),
+                    totalStats.queryTotal.count(),
+                    totalStats.fetchTotal.count(),
+                    totalStats.expandSearchTotal.count()
+                );
+            } else {
+                return SearchResponse.PhaseTook.NULL;
+            }
+        }
+
+        public SearchTimeProvider.RequestStatsHolder totalStats = new RequestStatsHolder();
+
+        public static final class RequestStatsHolder {
+            public CounterMetric dfsPreQueryTotal = new CounterMetric();
+            public CounterMetric canMatchTotal = new CounterMetric();
+            public CounterMetric queryTotal = new CounterMetric();
+            public CounterMetric fetchTotal = new CounterMetric();
+            public CounterMetric expandSearchTotal = new CounterMetric();
+        }
+
+        // call these to populate return values
+        public long getDFSPreQueryTotal() {
+            return totalStats.dfsPreQueryTotal.count();
+        }
+
+        public long getCanMatchTotal() {
+            return totalStats.canMatchTotal.count();
+        }
+
+        public long getQueryTotal() {
+            return totalStats.queryTotal.count();
+        }
+
+        public long getFetchTotal() {
+            return totalStats.fetchTotal.count();
+        }
+
+        public long getExpandSearchTotal() {
+            return totalStats.expandSearchTotal.count();
+        }
+
+        private void computeStats(Consumer<SearchTimeProvider.RequestStatsHolder> consumer) {
+            consumer.accept(totalStats);
+        }
+
+        @Override
+        public void onDFSPreQueryPhaseStart(SearchPhaseContext context) {}
+
+        @Override
+        public void onDFSPreQueryPhaseEnd(SearchPhaseContext context, long tookTime) {
+            computeStats(statsHolder -> { totalStats.dfsPreQueryTotal.inc(tookTime); });
+        }
+
+        @Override
+        public void onDFSPreQueryPhaseFailure(SearchPhaseContext context) {}
+
+        @Override
+        public void onCanMatchPhaseStart(SearchPhaseContext context) {}
+
+        @Override
+        public void onCanMatchPhaseEnd(SearchPhaseContext context, long tookTime) {
+            computeStats(statsHolder -> { totalStats.canMatchTotal.inc(tookTime); });
+        }
+
+        @Override
+        public void onCanMatchPhaseFailure(SearchPhaseContext context) {}
+
+        @Override
+        public void onQueryPhaseStart(SearchPhaseContext context) {}
+
+        @Override
+        public void onQueryPhaseEnd(SearchPhaseContext context, long tookTime) {
+            computeStats(statsHolder -> { totalStats.queryTotal.inc(tookTime); });
```

Comment:
Don't think incrementing tookTime looks right here, as this would be only called once at max, as SearchTimeProvider is created per request.
Same for others.